contracts/Dead.sol

```diff
@@ -42,16 +41,17 @@ contract Dead {
   @notice make sure to have approved this contract to transfer from your balance before calling this. Note that you cannot get these tokens out again unless you are the owner, or if you are the beneficiary and the owner has died
   @dev deposits ERC-20 tokens into the switch
   @param _tokenAddr the address of the ERC-20 token being deposited
+  @param _amount the number of tokens to deposit
   */
-  function depositERC20(address _tokenAddr) public {
+  function depositERC20(address _tokenAddr, uint _amount) public {
     EIP20 token = EIP20(_tokenAddr);
 
-    uint balanceToTransfer = token.balanceOf(msg.sender);
+    require(_amount <= token.balanceOf(msg.sender));
```

Comment:
Add another line after this: `require(_amount <= token.allowance(msg.sender, this))` to check that the depositor has approved this contract to transfer `_amount`, in addition to having the `_amount` themselves.

https://github.com/ConsenSys/Tokens/blob/master/contracts/eip20/EIP20Interface.sol#L42-L45